### DISCUSSION TITLE
Converting javascript array of object to python list of dictionaries take a lot time

### DISCUSSION THREAD

**Discussion by rheanV** [pyodide/pyodide#3901] [General]
I have a use case where I need to convert an array of objects to a list of dictionaries. The length of the array is greater than 350,000. It takes about 257-300 seconds to convert this array to Python by pyodide. Is there a way to decrease this conversion time?

**Comment by hoodmane** [comment]
Could you provide example code?

**Reply by rheanV** [reply]
Sure, this is the pyodide code that I run.
 `await pyodideRuntime.globals.set('data_info', data);
await pyodideRuntime.runPythonAsync('data = data_info.to_py()');`

Here, the data is an array of JSON objects. The length of this array can be anywhere from 350k to 3M.

**Reply by hoodmane** [reply]
Are you loading the data from a string? In that case you could consider using `import json; json.loads(serialized_data)`. The `to_py` converter has to do a significant amount of extra work to handle cycles, if you know that the data is acyclic then you can probably write your own code that will be faster.

**Reply by rheanV** [reply]
Ahh, that worked. Silly of me not to try that earlier! Thank you so much :)

**Reply by rheanV** [reply]
Following up, is there is a way to reduce time for pyodideRuntime.globals.set() method? It takes about 10 seconds to set the string. Here is the code:
`const dataString = JSON.stringify(data);
await pyodideRuntime.globals.set('data_info', dataString);
await pyodideRuntime.runPythonAsync('import json;final_data = json.loads(data_info)')`
